src/ripe_rainbow/domain/base/assertions.py

```diff
@@ -100,7 +116,8 @@ def has_text(self, selector, text, ensure = False):
             # from an interactable point of view
             if ensure: self.driver.safe(self.driver.ensure_visible, element)
 
-            if not element.text == text:
+            element_text = element.get_attribute("value") if is_input else element.text

```

Comment:
No parameter, grab the element check if it of type input and if that's the case assume that it's an input.
Less parameters == better API

Reply:
🤦‍♂ , that's so much better and obvious.